common/predictive-text/worker/correction/distance-modeler.ts

```diff
@@ -54,15 +62,17 @@ namespace correction {
     }
 
     get inputSamplingCost(): number {
-      // TODO:  Optimize so that we're not frequently recomputing this?
-      // TODO:  We might should generalize this so that the probability-to-cost function isn't directly hard-coded.
-      //        Seems like a decent first conversion function though, at least.
-
-      let THRESHOLD = 0.001;
-      // Should technically re-normalize the sampling distribution.
-      // We use -ln(p) because a positive cost is worse.  ln(p) is always <= 0. 
-      // TODO:  probably more efficient to instead use actual probability space... but that'll involve extra changes.
-      return this.priorInput.map(mass => mass.p > THRESHOLD ? mass.p : THRESHOLD).reduce((previous, current) => previous - Math.log(current), 0);
+      if(this._inputCost !== undefined) {
+        return this._inputCost;
+      } else {
+        let MIN_P = 0.0001;
```

Comment:
Could this be extracted as a `const` to an outer scope? And perhaps given a more descriptive name?

Reply:
"Could"?  Sure.  But in what manner?  Would a static, read-only class field be fine?

Now that I'm looking at it a bit later, something like "EPSILON" would probably be a bit more standard.  I mean, the whole reason it exists there is to prevent the `log(0) == -Infinity` problem, and "epsilon" is the standard math symbol for use in such cases.

Though, in this case it does come with a bit of extra interpretation.

Reply:
Yes, this works! And a comment explaining that this is to avoid a zero probability black hole! 😄